### DISCUSSION TITLE
Embargo conf requires a login - we do not have one

### DISCUSSION THREAD

**Discussion by ctsdownloads** [fwupd/fwupd#6453] [Q&A] [answered]
**Describe the bug**
Create an embargo conf, upon fresh, getting Access denied server error.
**Steps to Reproduce**
Create /etc/fwupd/remotes.d/our-embargo.conf, drop in:
```
[fwupd Remote]
Enabled=true
Title=Embargoed for Us
#Keyring=gpg
MetadataURI=TheExpectedURLhere.xml.gz
ReportURI=https://fwupd.org/lvfs/firmware/report
OrderBefore=lvfs,fwupd
```
Also tried:

```
[fwupd Remote]
Enabled=true
Title=Embargoed for Us
#Keyring=gpg
MetadataURI=TheExpectedURLhere.xml.gz
ReportURI=https://fwupd.org/lvfs/firmware/report
OrderBefore=lvfs,fwupd
#Username=
#Password=
```

Save the file, run fwupdmgr refresh or  fwupdmgr refresh --force as we have on previous releases.
**Expected behavior**
Successfully run fwupdmgr refresh without server errors.
**fwupd version information**
Please provide the version of the daemon and client.

On Fedora 39, fully updated:
```shell
fwupdmgr 1.9.9
```

Please note how you installed it (`apt`, `dnf`, `pacman`, source, etc):
`dnf`

<details>

<summary>**fwupd device information**</summary>

Please provide the output of the fwupd devices recognized in your system.

```shell
fwupdmgr get-devices --show-all-devices
```
N/A
</details>

**Additional questions**

- Operating system and version: Fedora 39
- Have you tried rebooting? Yes
- Is this a regression? No, appears to be a new feature. Previous to 1.9.9, our embargo conf worked fine. It's the attempt at authorizing a login that is causing the issue. Ideally, we'd like to see a flag to skip this.

**Comment by hughsie** [comment, marked as answer]
Hi! Sorry about this. I sent an email announcing the changes here: https://lists.linuxfoundation.org/pipermail/lvfs-announce/2023-November/000087.html

We can opt out your vendor (Framework?) if you want, but I'd much rather `nrp`, `ktl` or `dhs` create you an LVFS account so that you can use your username+passtoken -- as it's much more secure. Let me know what you'd like to do.

**Comment by superm1** [comment]
The side effect of going authentication is going to be that any partners you've shared the embargo conf need to have an explicit relationship described in LVFS in order to access the repo.

@hughsie can LVFS actually express an IHV relationship yet?

**Comment by hughsie** [comment]
> can LVFS actually express an IHV relationship yet

Yup, you can even have multiple ODMs for one component now.

**Comment by ctsdownloads** [comment]
@hughsie @superm1 Appreciate the insights. I have looped in `ktl` via email and expect to hear back from them on how we'd want to approach this from their side of things.

**Comment by kiram9** [comment]
@hughsie I would like to go with getting our partners using a relationship as you recommend. However a few questions. 
1. in the admin console, for a specific firmware, I assume we would provide access to a vendor in the firmware->Vendor tab and add our vendor here. 

It looks like there are vendor specific permissions in vendor -> affiliations.

So to properly limit this so our partner cannot modify our release, but can view it, we would create a relationship, and then when we upload the firmware we would add the vendor affilitiation in the firmware upload.

For permissions under vendor->affiliations, i do not see a permission for the vendor to access embargoed firmware. I assume we could revoke all permissions for the vendor, and they would be able to access it under embargo as long as we share that firmware with them. 
Basically I want read only access to embargo for vendors for specific firmware uploads. 

Can you add AMD as an affiliate for us?

**Comment by hughsie** [comment]
> I assume we would provide access to a vendor in the firmware->Vendor tab and add our vendor here.

Exactly that.

> For permissions under vendor->affiliations, i do not see a permission for the vendor to access embargoed firmware

This is the "View firmware" ACL.

> Can you add AMD as an affiliate for us?

Done.

>  Basically I want read only access to embargo for vendors for specific firmware uploads.

The way it works under the hood is that any firmwares you explicitly share with AMD appear in **their** embargo metadata as well as yours.

**Reply by superm1** [reply]
>> Can you add AMD as an affiliate for us?

> Done.

Hmm I just looked and I don't see it.
<img width="615" alt="image" src="https://github.com/fwupd/fwupd/assets/494526/fda9144f-61bd-431e-a600-91e0c63249e1">

**Reply by hughsie** [reply]
Nah, it's the other way around -- Framework can share firmware with AMD but AMD can't share with Framework. You can do upload *for* Framework, but the View firmware ACL doesn't confer that ability. If that's wrong let me know.

**Reply by superm1** [reply]
I double checked the AMD embargo channel and don't see any Framework content.

**Reply by hughsie** [reply]
@kiram9 have you shared a firmware with AMD?

**Reply by ctsdownloads** [reply]
@hughsie I will intercept this and see what the plan is on my end. @superm1 I'll get with @kiram9 and figure out the plan.

**Reply by kiram9** [reply]
@superm1 I just uploaded and added AMD as the vendor. It might show up in an hour or so.
